visionmetrics/key_value_pair/key_value_pair_eval.py

```diff
@@ -0,0 +1,314 @@
+from enum import Enum
+import logging
+
+from visionmetrics.key_value_pair.key_value_pair_eval_base import KeyValuePairEvaluatorBase, SupportedKeyWiseMetric
+
+logger = logging.getLogger(__name__)
+
+
+class JSONSchemaKeyType(str, Enum):
+    String = "string"
+    Number = "number"
+    Integer = "integer"
+    Boolean = "boolean"
+    Array = "array"
+    Object = "object"
+
+
+# Constants for schemas
+SIMPLE_KEY_TYPES = [JSONSchemaKeyType.String, JSONSchemaKeyType.Number, JSONSchemaKeyType.Integer, JSONSchemaKeyType.Boolean]
+OUT_OF_DISTRIBUTION_ENUM_KEY = "<|other|>"
+
+# Constants for predictions and targets
+VALUE_SUBKEY = "value"
+GROUNDINGS_SUBKEY = "groundings"
+DESCRIPTION_SUBKEY = "description"
+
+
+class KeyValuePairExtractionScore(KeyValuePairEvaluatorBase):
+    """
+    Evaluator for a key-value pair dataset. The default evaluator (if the key does not have values of a particular form) is the caption.AzureOpenAITextModelCategoricalScore due to its flexibility.
```

Comment:
Good to forget all your existing knowledge about vision-datasets and see if this documentation still makes sense to determine whether the documentation is helpful or not

Reply:
Good point. Added more context and information to describe the classes better.